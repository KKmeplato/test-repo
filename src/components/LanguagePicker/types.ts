export interface LanguagePickerProps {
  locales: string[];
  selected: string;
  onSelect?: (locale: string) => void;
}
