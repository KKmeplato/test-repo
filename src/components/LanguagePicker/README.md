### LanguagePicker

```tsx
<LanguagePicker
  locales={["en", "de", "es", "pt-BR"]}
  selected="de"
  onSelect={(locale) => {}}
/>
```

- [x] Meplato light theme
- [x] Meplato dark theme
- [x] Webridge light theme
- [x] Webridge dark theme
