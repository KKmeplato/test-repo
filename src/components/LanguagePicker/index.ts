export * from "./LanguagePicker";
export * from "./types";
