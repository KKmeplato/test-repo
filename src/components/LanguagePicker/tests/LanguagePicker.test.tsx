import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { expect } from "vitest";

import { LanguagePicker } from "../LanguagePicker";

function Wrapper({ children }: { children: React.ReactElement }) {
  return <div id="uikit-theme">{children}</div>;
}

describe("<LanguagePicker />", () => {
  it("renders", () => {
    const result = render(
      <LanguagePicker selected="de" locales={["en", "de", "es", "pt-BR"]} />,
      { wrapper: Wrapper }
    );

    expect(result).toMatchSnapshot();
  });

  it("renders when clicked", async () => {
    const result = render(
      <LanguagePicker selected="de" locales={["en", "de", "es", "pt-BR"]} />,
      { wrapper: Wrapper }
    );

    await userEvent.click(result.getByRole("button"));

    expect(result).toMatchSnapshot();
  });

  it("opens an ActionList with the languages when clicked", async () => {
    const { getByRole, getByText } = render(
      <LanguagePicker selected="en" locales={["en", "de", "es", "pt-BR"]} />,
      { wrapper: Wrapper }
    );

    await userEvent.click(getByRole("button"));

    expect(getByText("Spanish")).toBeDefined();
    expect(getByText("German")).toBeDefined();
    expect(getByText("Brazilian Portuguese")).toBeDefined();
  });

  it("responds to onSelect when a language item is clicked", async () => {
    const mockOnSelect = vi.fn();
    const { getByRole, getAllByRole } = render(
      <LanguagePicker
        selected="en"
        locales={["en", "de", "es", "pt-BR"]}
        onSelect={mockOnSelect}
      />,
      { wrapper: Wrapper }
    );

    await userEvent.click(getByRole("button"));
    await userEvent.click(getAllByRole("menuitem")[2]);

    expect(mockOnSelect).toHaveBeenCalledWith("es");
  });
});
