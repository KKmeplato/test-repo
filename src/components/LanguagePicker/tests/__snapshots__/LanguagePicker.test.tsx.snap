// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<LanguagePicker /> > renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="uikit-theme"
      >
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="rounded enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1 enabled:hover:bg-surface-neutral-hovered enabled:active:bg-surface-neutral-pressed"
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          <span
            class="flex items-center justify-center gap-2 pl-0.5 pr-1"
          >
            <svg
              class="h-5 py-0.5"
              fill="none"
              viewBox="0 0 27 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 2.66667C0 1.19583 1.19583 0 2.66667 0H10.6667H12.6667H13.3333H24C25.4708 0 26.6667 1.19583 26.6667 2.66667V13.3333C26.6667 14.8042 25.4708 16 24 16H13.3333H12.6667H10.6667H2.66667C1.19583 16 0 14.8042 0 13.3333V2.66667ZM13.3333 2.66667V13.3333H24V2.66667H13.3333ZM7.42917 4.6625C7.29583 4.3625 6.99583 4.16667 6.66667 4.16667C6.3375 4.16667 6.0375 4.3625 5.90417 4.6625L3.2375 10.6625C3.05 11.0833 3.24167 11.575 3.6625 11.7625C4.08333 11.95 4.575 11.7583 4.7625 11.3375L5.13333 10.5H8.2L8.57083 11.3375C8.75833 11.7583 9.25 11.9458 9.67083 11.7625C10.0917 11.5792 10.2792 11.0833 10.0958 10.6625L7.42917 4.6625ZM6.66667 7.05L7.45833 8.83333H5.875L6.66667 7.05ZM18.6667 4.16667C19.125 4.16667 19.5 4.54167 19.5 5V5.16667H21.3333H22C22.4583 5.16667 22.8333 5.54167 22.8333 6C22.8333 6.45833 22.4583 6.83333 22 6.83333H21.9167L21.85 7.02083C21.4792 8.0375 20.9167 8.9625 20.2 9.74583C20.2375 9.77083 20.275 9.79167 20.3125 9.8125L21.1 10.2833C21.4958 10.5208 21.6208 11.0333 21.3875 11.425C21.1542 11.8167 20.6375 11.9458 20.2458 11.7125L19.4583 11.2417C19.2708 11.1292 19.0917 11.0125 18.9125 10.8875C18.4708 11.2 18 11.4708 17.4958 11.6958L17.3458 11.7625C16.925 11.95 16.4333 11.7583 16.2458 11.3375C16.0583 10.9167 16.25 10.425 16.6708 10.2375L16.8208 10.1708C17.0875 10.05 17.3458 9.91667 17.5917 9.7625L17.0833 9.25417C16.7583 8.92917 16.7583 8.4 17.0833 8.075C17.4083 7.75 17.9375 7.75 18.2625 8.075L18.8708 8.68333L18.8917 8.70417C19.4083 8.15833 19.8292 7.525 20.1333 6.82917H18.6667H15.6667C15.2083 6.82917 14.8333 6.45417 14.8333 5.99583C14.8333 5.5375 15.2083 5.1625 15.6667 5.1625H17.8333V4.99583C17.8333 4.5375 18.2083 4.1625 18.6667 4.1625V4.16667Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="font-mono text-sm font-medium uppercase"
            >
              de
            </span>
          </span>
          <span
            class="sr-only"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="uikit-theme"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        class="rounded enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1 enabled:hover:bg-surface-neutral-hovered enabled:active:bg-surface-neutral-pressed"
        data-state="closed"
        id="radix-:r0:"
        type="button"
      >
        <span
          class="flex items-center justify-center gap-2 pl-0.5 pr-1"
        >
          <svg
            class="h-5 py-0.5"
            fill="none"
            viewBox="0 0 27 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 2.66667C0 1.19583 1.19583 0 2.66667 0H10.6667H12.6667H13.3333H24C25.4708 0 26.6667 1.19583 26.6667 2.66667V13.3333C26.6667 14.8042 25.4708 16 24 16H13.3333H12.6667H10.6667H2.66667C1.19583 16 0 14.8042 0 13.3333V2.66667ZM13.3333 2.66667V13.3333H24V2.66667H13.3333ZM7.42917 4.6625C7.29583 4.3625 6.99583 4.16667 6.66667 4.16667C6.3375 4.16667 6.0375 4.3625 5.90417 4.6625L3.2375 10.6625C3.05 11.0833 3.24167 11.575 3.6625 11.7625C4.08333 11.95 4.575 11.7583 4.7625 11.3375L5.13333 10.5H8.2L8.57083 11.3375C8.75833 11.7583 9.25 11.9458 9.67083 11.7625C10.0917 11.5792 10.2792 11.0833 10.0958 10.6625L7.42917 4.6625ZM6.66667 7.05L7.45833 8.83333H5.875L6.66667 7.05ZM18.6667 4.16667C19.125 4.16667 19.5 4.54167 19.5 5V5.16667H21.3333H22C22.4583 5.16667 22.8333 5.54167 22.8333 6C22.8333 6.45833 22.4583 6.83333 22 6.83333H21.9167L21.85 7.02083C21.4792 8.0375 20.9167 8.9625 20.2 9.74583C20.2375 9.77083 20.275 9.79167 20.3125 9.8125L21.1 10.2833C21.4958 10.5208 21.6208 11.0333 21.3875 11.425C21.1542 11.8167 20.6375 11.9458 20.2458 11.7125L19.4583 11.2417C19.2708 11.1292 19.0917 11.0125 18.9125 10.8875C18.4708 11.2 18 11.4708 17.4958 11.6958L17.3458 11.7625C16.925 11.95 16.4333 11.7583 16.2458 11.3375C16.0583 10.9167 16.25 10.425 16.6708 10.2375L16.8208 10.1708C17.0875 10.05 17.3458 9.91667 17.5917 9.7625L17.0833 9.25417C16.7583 8.92917 16.7583 8.4 17.0833 8.075C17.4083 7.75 17.9375 7.75 18.2625 8.075L18.8708 8.68333L18.8917 8.70417C19.4083 8.15833 19.8292 7.525 20.1333 6.82917H18.6667H15.6667C15.2083 6.82917 14.8333 6.45417 14.8333 5.99583C14.8333 5.5375 15.2083 5.1625 15.6667 5.1625H17.8333V4.99583C17.8333 4.5375 18.2083 4.1625 18.6667 4.1625V4.16667Z"
              fill="currentColor"
            />
          </svg>
          <span
            class="font-mono text-sm font-medium uppercase"
          >
            de
          </span>
        </span>
        <span
          class="sr-only"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LanguagePicker /> > renders when clicked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <span
      data-radix-focus-guard=""
      style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
      tabindex="0"
    />
    <div>
      <div
        id="uikit-theme"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="true"
          aria-haspopup="menu"
          class="rounded enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1 enabled:hover:bg-surface-neutral-hovered enabled:active:bg-surface-neutral-pressed bg-surface-neutral-pressed"
          data-state="open"
          id="radix-:r2:"
          type="button"
        >
          <span
            class="flex items-center justify-center gap-2 pl-0.5 pr-1"
          >
            <svg
              class="h-5 py-0.5"
              fill="none"
              viewBox="0 0 27 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 2.66667C0 1.19583 1.19583 0 2.66667 0H10.6667H12.6667H13.3333H24C25.4708 0 26.6667 1.19583 26.6667 2.66667V13.3333C26.6667 14.8042 25.4708 16 24 16H13.3333H12.6667H10.6667H2.66667C1.19583 16 0 14.8042 0 13.3333V2.66667ZM13.3333 2.66667V13.3333H24V2.66667H13.3333ZM7.42917 4.6625C7.29583 4.3625 6.99583 4.16667 6.66667 4.16667C6.3375 4.16667 6.0375 4.3625 5.90417 4.6625L3.2375 10.6625C3.05 11.0833 3.24167 11.575 3.6625 11.7625C4.08333 11.95 4.575 11.7583 4.7625 11.3375L5.13333 10.5H8.2L8.57083 11.3375C8.75833 11.7583 9.25 11.9458 9.67083 11.7625C10.0917 11.5792 10.2792 11.0833 10.0958 10.6625L7.42917 4.6625ZM6.66667 7.05L7.45833 8.83333H5.875L6.66667 7.05ZM18.6667 4.16667C19.125 4.16667 19.5 4.54167 19.5 5V5.16667H21.3333H22C22.4583 5.16667 22.8333 5.54167 22.8333 6C22.8333 6.45833 22.4583 6.83333 22 6.83333H21.9167L21.85 7.02083C21.4792 8.0375 20.9167 8.9625 20.2 9.74583C20.2375 9.77083 20.275 9.79167 20.3125 9.8125L21.1 10.2833C21.4958 10.5208 21.6208 11.0333 21.3875 11.425C21.1542 11.8167 20.6375 11.9458 20.2458 11.7125L19.4583 11.2417C19.2708 11.1292 19.0917 11.0125 18.9125 10.8875C18.4708 11.2 18 11.4708 17.4958 11.6958L17.3458 11.7625C16.925 11.95 16.4333 11.7583 16.2458 11.3375C16.0583 10.9167 16.25 10.425 16.6708 10.2375L16.8208 10.1708C17.0875 10.05 17.3458 9.91667 17.5917 9.7625L17.0833 9.25417C16.7583 8.92917 16.7583 8.4 17.0833 8.075C17.4083 7.75 17.9375 7.75 18.2625 8.075L18.8708 8.68333L18.8917 8.70417C19.4083 8.15833 19.8292 7.525 20.1333 6.82917H18.6667H15.6667C15.2083 6.82917 14.8333 6.45417 14.8333 5.99583C14.8333 5.5375 15.2083 5.1625 15.6667 5.1625H17.8333V4.99583C17.8333 4.5375 18.2083 4.1625 18.6667 4.1625V4.16667Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="font-mono text-sm font-medium uppercase"
            >
              de
            </span>
          </span>
          <span
            class="sr-only"
          />
        </button>
        <div
          data-radix-popper-content-wrapper=""
          dir="ltr"
          style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 0px); min-width: max-content; --radix-popper-available-width: 0px; --radix-popper-available-height: 0px; --radix-popper-anchor-width: 0px; --radix-popper-anchor-height: 0px; --radix-popper-transform-origin: 0% 0px;"
        >
          <div
            aria-labelledby="radix-:r2:"
            aria-orientation="vertical"
            class="flex flex-col overflow-hidden rounded-level-1 bg-surface py-2 shadow-md data-[state=closed]:animate-dropdown-close data-[state=open]:animate-dropdown-open max-w-max z-dropdown mt-1"
            data-align="start"
            data-orientation="vertical"
            data-radix-menu-content=""
            data-side="bottom"
            data-state="open"
            dir="ltr"
            id="radix-:r3:"
            role="menu"
            style="outline: none; --radix-dropdown-menu-content-transform-origin: var(--radix-popper-transform-origin); --radix-dropdown-menu-content-available-width: var(--radix-popper-available-width); --radix-dropdown-menu-content-available-height: var(--radix-popper-available-height); --radix-dropdown-menu-trigger-width: var(--radix-popper-anchor-width); --radix-dropdown-menu-trigger-height: var(--radix-popper-anchor-height);"
            tabindex="-1"
          >
            <div
              class="relative px-2"
            >
              <button
                class="flex group w-full items-start gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed"
                data-orientation="vertical"
                data-radix-collection-item=""
                role="menuitem"
                tabindex="-1"
              >
                <span
                  class="flex gap-4"
                >
                  <span
                    class="text-sm text-subdued"
                  >
                    <span
                      class="uppercase"
                    >
                      en
                    </span>
                  </span>
                  <span
                    class="flex flex-col items-start text-sm font-normal"
                  >
                    <span>
                      Englisch
                    </span>
                    <span
                      class="text-subdued"
                    >
                      English
                    </span>
                  </span>
                </span>
              </button>
            </div>
            <div
              class="relative px-2"
            >
              <div
                class="absolute left-0 h-full w-[3px] rounded bg-interactive-primary"
              />
              <button
                class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:bg-surface-selected enabled:hover:bg-surface-selected-hovered enabled:active:bg-surface-selected-pressed items-center"
                data-orientation="vertical"
                data-radix-collection-item=""
                role="menuitem"
                tabindex="-1"
              >
                <span
                  class="flex gap-4"
                >
                  <span
                    class="text-sm text-subdued"
                  >
                    <span
                      class="uppercase"
                    >
                      de
                    </span>
                  </span>
                  <span
                    class="flex flex-col items-start text-sm font-normal"
                  >
                    <span>
                      Deutsch
                    </span>
                  </span>
                </span>
                <span
                  class="w-5 h-5 flex-shrink-0 group-disabled:text-icon-disabled text-icon group-enabled:text-interactive-primary group-enabled:group-hover:text-interactive-primary-hovered"
                >
                  <svg
                    class="text-interactive-primary"
                    fill="none"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M17.6652 4.62025C18.1116 5.06662 18.1116 5.79154 17.6652 6.23792L8.52338 15.3798C8.077 15.8261 7.35208 15.8261 6.9057 15.3798L2.33478 10.8088C1.88841 10.3625 1.88841 9.63754 2.33478 9.19116C2.78116 8.74478 3.50608 8.74478 3.95246 9.19116L7.71633 12.9515L16.0511 4.62025C16.4975 4.17387 17.2224 4.17387 17.6688 4.62025H17.6652Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="relative px-2"
            >
              <button
                class="flex group w-full items-start gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed"
                data-orientation="vertical"
                data-radix-collection-item=""
                role="menuitem"
                tabindex="-1"
              >
                <span
                  class="flex gap-4"
                >
                  <span
                    class="text-sm text-subdued"
                  >
                    <span
                      class="uppercase"
                    >
                      es
                    </span>
                  </span>
                  <span
                    class="flex flex-col items-start text-sm font-normal"
                  >
                    <span>
                      Spanisch
                    </span>
                    <span
                      class="text-subdued"
                    >
                      español
                    </span>
                  </span>
                </span>
              </button>
            </div>
            <div
              class="relative px-2"
            >
              <button
                class="flex group w-full items-start gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed"
                data-orientation="vertical"
                data-radix-collection-item=""
                role="menuitem"
                tabindex="-1"
              >
                <span
                  class="flex gap-4"
                >
                  <span
                    class="text-sm text-subdued"
                  >
                    <span
                      class="uppercase"
                    >
                      pt
                    </span>
                  </span>
                  <span
                    class="flex flex-col items-start text-sm font-normal"
                  >
                    <span>
                      Portugiesisch (Brasilien)
                    </span>
                    <span
                      class="text-subdued"
                    >
                      português (Brasil)
                    </span>
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      data-radix-focus-guard=""
      style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
      tabindex="0"
    />
  </body>,
  "container": <div>
    <div
      id="uikit-theme"
    >
      <button
        aria-controls="radix-:r3:"
        aria-expanded="true"
        aria-haspopup="menu"
        class="rounded enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1 enabled:hover:bg-surface-neutral-hovered enabled:active:bg-surface-neutral-pressed bg-surface-neutral-pressed"
        data-state="open"
        id="radix-:r2:"
        type="button"
      >
        <span
          class="flex items-center justify-center gap-2 pl-0.5 pr-1"
        >
          <svg
            class="h-5 py-0.5"
            fill="none"
            viewBox="0 0 27 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 2.66667C0 1.19583 1.19583 0 2.66667 0H10.6667H12.6667H13.3333H24C25.4708 0 26.6667 1.19583 26.6667 2.66667V13.3333C26.6667 14.8042 25.4708 16 24 16H13.3333H12.6667H10.6667H2.66667C1.19583 16 0 14.8042 0 13.3333V2.66667ZM13.3333 2.66667V13.3333H24V2.66667H13.3333ZM7.42917 4.6625C7.29583 4.3625 6.99583 4.16667 6.66667 4.16667C6.3375 4.16667 6.0375 4.3625 5.90417 4.6625L3.2375 10.6625C3.05 11.0833 3.24167 11.575 3.6625 11.7625C4.08333 11.95 4.575 11.7583 4.7625 11.3375L5.13333 10.5H8.2L8.57083 11.3375C8.75833 11.7583 9.25 11.9458 9.67083 11.7625C10.0917 11.5792 10.2792 11.0833 10.0958 10.6625L7.42917 4.6625ZM6.66667 7.05L7.45833 8.83333H5.875L6.66667 7.05ZM18.6667 4.16667C19.125 4.16667 19.5 4.54167 19.5 5V5.16667H21.3333H22C22.4583 5.16667 22.8333 5.54167 22.8333 6C22.8333 6.45833 22.4583 6.83333 22 6.83333H21.9167L21.85 7.02083C21.4792 8.0375 20.9167 8.9625 20.2 9.74583C20.2375 9.77083 20.275 9.79167 20.3125 9.8125L21.1 10.2833C21.4958 10.5208 21.6208 11.0333 21.3875 11.425C21.1542 11.8167 20.6375 11.9458 20.2458 11.7125L19.4583 11.2417C19.2708 11.1292 19.0917 11.0125 18.9125 10.8875C18.4708 11.2 18 11.4708 17.4958 11.6958L17.3458 11.7625C16.925 11.95 16.4333 11.7583 16.2458 11.3375C16.0583 10.9167 16.25 10.425 16.6708 10.2375L16.8208 10.1708C17.0875 10.05 17.3458 9.91667 17.5917 9.7625L17.0833 9.25417C16.7583 8.92917 16.7583 8.4 17.0833 8.075C17.4083 7.75 17.9375 7.75 18.2625 8.075L18.8708 8.68333L18.8917 8.70417C19.4083 8.15833 19.8292 7.525 20.1333 6.82917H18.6667H15.6667C15.2083 6.82917 14.8333 6.45417 14.8333 5.99583C14.8333 5.5375 15.2083 5.1625 15.6667 5.1625H17.8333V4.99583C17.8333 4.5375 18.2083 4.1625 18.6667 4.1625V4.16667Z"
              fill="currentColor"
            />
          </svg>
          <span
            class="font-mono text-sm font-medium uppercase"
          >
            de
          </span>
        </span>
        <span
          class="sr-only"
        />
      </button>
      <div
        data-radix-popper-content-wrapper=""
        dir="ltr"
        style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 0px); min-width: max-content; --radix-popper-available-width: 0px; --radix-popper-available-height: 0px; --radix-popper-anchor-width: 0px; --radix-popper-anchor-height: 0px; --radix-popper-transform-origin: 0% 0px;"
      >
        <div
          aria-labelledby="radix-:r2:"
          aria-orientation="vertical"
          class="flex flex-col overflow-hidden rounded-level-1 bg-surface py-2 shadow-md data-[state=closed]:animate-dropdown-close data-[state=open]:animate-dropdown-open max-w-max z-dropdown mt-1"
          data-align="start"
          data-orientation="vertical"
          data-radix-menu-content=""
          data-side="bottom"
          data-state="open"
          dir="ltr"
          id="radix-:r3:"
          role="menu"
          style="outline: none; --radix-dropdown-menu-content-transform-origin: var(--radix-popper-transform-origin); --radix-dropdown-menu-content-available-width: var(--radix-popper-available-width); --radix-dropdown-menu-content-available-height: var(--radix-popper-available-height); --radix-dropdown-menu-trigger-width: var(--radix-popper-anchor-width); --radix-dropdown-menu-trigger-height: var(--radix-popper-anchor-height);"
          tabindex="-1"
        >
          <div
            class="relative px-2"
          >
            <button
              class="flex group w-full items-start gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed"
              data-orientation="vertical"
              data-radix-collection-item=""
              role="menuitem"
              tabindex="-1"
            >
              <span
                class="flex gap-4"
              >
                <span
                  class="text-sm text-subdued"
                >
                  <span
                    class="uppercase"
                  >
                    en
                  </span>
                </span>
                <span
                  class="flex flex-col items-start text-sm font-normal"
                >
                  <span>
                    Englisch
                  </span>
                  <span
                    class="text-subdued"
                  >
                    English
                  </span>
                </span>
              </span>
            </button>
          </div>
          <div
            class="relative px-2"
          >
            <div
              class="absolute left-0 h-full w-[3px] rounded bg-interactive-primary"
            />
            <button
              class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:bg-surface-selected enabled:hover:bg-surface-selected-hovered enabled:active:bg-surface-selected-pressed items-center"
              data-orientation="vertical"
              data-radix-collection-item=""
              role="menuitem"
              tabindex="-1"
            >
              <span
                class="flex gap-4"
              >
                <span
                  class="text-sm text-subdued"
                >
                  <span
                    class="uppercase"
                  >
                    de
                  </span>
                </span>
                <span
                  class="flex flex-col items-start text-sm font-normal"
                >
                  <span>
                    Deutsch
                  </span>
                </span>
              </span>
              <span
                class="w-5 h-5 flex-shrink-0 group-disabled:text-icon-disabled text-icon group-enabled:text-interactive-primary group-enabled:group-hover:text-interactive-primary-hovered"
              >
                <svg
                  class="text-interactive-primary"
                  fill="none"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17.6652 4.62025C18.1116 5.06662 18.1116 5.79154 17.6652 6.23792L8.52338 15.3798C8.077 15.8261 7.35208 15.8261 6.9057 15.3798L2.33478 10.8088C1.88841 10.3625 1.88841 9.63754 2.33478 9.19116C2.78116 8.74478 3.50608 8.74478 3.95246 9.19116L7.71633 12.9515L16.0511 4.62025C16.4975 4.17387 17.2224 4.17387 17.6688 4.62025H17.6652Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="relative px-2"
          >
            <button
              class="flex group w-full items-start gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed"
              data-orientation="vertical"
              data-radix-collection-item=""
              role="menuitem"
              tabindex="-1"
            >
              <span
                class="flex gap-4"
              >
                <span
                  class="text-sm text-subdued"
                >
                  <span
                    class="uppercase"
                  >
                    es
                  </span>
                </span>
                <span
                  class="flex flex-col items-start text-sm font-normal"
                >
                  <span>
                    Spanisch
                  </span>
                  <span
                    class="text-subdued"
                  >
                    español
                  </span>
                </span>
              </span>
            </button>
          </div>
          <div
            class="relative px-2"
          >
            <button
              class="flex group w-full items-start gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed"
              data-orientation="vertical"
              data-radix-collection-item=""
              role="menuitem"
              tabindex="-1"
            >
              <span
                class="flex gap-4"
              >
                <span
                  class="text-sm text-subdued"
                >
                  <span
                    class="uppercase"
                  >
                    pt
                  </span>
                </span>
                <span
                  class="flex flex-col items-start text-sm font-normal"
                >
                  <span>
                    Portugiesisch (Brasilien)
                  </span>
                  <span
                    class="text-subdued"
                  >
                    português (Brasil)
                  </span>
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
